lib/scorecalculator.rb

```diff
@@ -0,0 +1,76 @@
+class ScoreCalculator
+  def initialize
+    @sheet = []
+  end
+
+  def input(*args)
+    errors(*args)
+    @sheet << args
+    bonuspoints
+    puts "You scored #{args.sum} for frame #{currentframe}. Your overall score so far is #{currentscore}"
+    specialannouncement
+    reset if completegame?
+  end
+
+  def scoreboard
+    score = 0
+    @sheet.map.with_index do |x, i|
+      puts "Frame #{i + 1}: #{x.flatten} | #{x.flatten.sum} | #{score += x.flatten.sum}"
```

Comment:
nice. I wonder if it would make sense to extract the scoreboard into a separate class?